src/ui/layout/tutorial.md

```diff
@@ -1,237 +1,390 @@
+@@ -23 +19,6 @@
++            children: [
++              TitleSection(
++                name: 'Oeschinen Lake Campground',
++                location: 'Kandersteg, Switzerland',
++              ),
++            ],
 ```
 
 {{site.alert.tip}}
-  * When pasting code into your app, indentation can
-    become skewed. You can fix this in your Flutter editor
-    using the [automatic reformatting support][].
-  * For a faster development experience,
-    try Flutter's [hot reload][] feature.
-  * If you have problems, compare your code to [`lib/main.dart`][].
+
+* When pasting code into your app, indentation can become skewed.
+      To fix this in your Flutter editor, use [automatic reformatting support][].
+* To accelerate your development, try Flutter's [hot reload][] feature.
+* If you have problems, compare your code to [`lib/main.dart`][].
+
 {{site.alert.end}}
 
-## Step 3: Implement the button row
+## Add the Button section
+
+In this section, add the buttons that will add functionality to your app.
 
 <?code-excerpt path-base="layout/lakes/step3"?>
 
-The button section contains 3 columns that use the same
-layout&mdash;an icon over a row of text.
-The columns in this row are evenly spaced,
-and the text and icons are painted with the primary color.
+The **Button** section contains three columns that use the same layout:
+an icon over a row of text.
+
+{% include docs/app-figure.md
+   image="ui/layout/layout-sketch-button-block-unlabeled.svg"
+   caption="The Button section as sketch and prototype UI" %}
 
-Since the code for building each column is almost identical,
-create a private helper method named `buildButtonColumn()`,
-which takes a color, an `Icon` and `Text`,
-and returns a column with its widgets painted in the given color.
+Plan to distribute these columns in one row so each takes the same
+amount of space. Paint all text and icons with the primary color.
 
-<?code-excerpt "lib/main.dart (_buildButtonColumn)" title?>
+### Add the `ButtonSection` widget
+
+Add the following code after the `TitleSection` widget to contain the code
+to build the row of buttons.
+
+<?code-excerpt "lib/main.dart (ButtonStart)" title?>
 ```dart
-class MyApp extends StatelessWidget {
-  const MyApp({super.key});
+class ButtonSection extends StatelessWidget {
+  const ButtonSection({super.key});
 
   @override
   Widget build(BuildContext context) {
-    // ···
+    final Color color = Theme.of(context).primaryColor;
+// ···
   }
+}
+```
 
-  Column _buildButtonColumn(Color color, IconData icon, String label) {
+### Create a widget to make buttons
+
+As the code for each column could use the same syntax,
+create a widget named `BuildButtonColumn()`.
+Each call to this method passes a color, an `Icon` widget and a `Text` widget.
+Given these parameters, the helper returns a `Column` widget.
+Each contains two children: the `Icon` and the `Text`.
+A `Padding` widget separates these children.
+The helper also paints these children in the given color.
```

Comment:
`ButtonColumn` is a widget (with a constructor, fields, and a build method) rather than a helper method, so we need to update this text to reflect that. How about something like the following?

```suggestion
The constructor of the widget accepts a color,
icon data, and a label for the button.
Using these values, the widget builds a `Column`
with an `Icon` and a stylized `Text` widget as its children.
To help separate these children, the `Text` widget
is wrapped with a `Padding` widget.
```